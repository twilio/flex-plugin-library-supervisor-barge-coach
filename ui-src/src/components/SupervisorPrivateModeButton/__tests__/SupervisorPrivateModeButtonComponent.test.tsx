import React from 'react';
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import userEvent from '@testing-library/user-event';
import BargeCoachService from '../../../utils/serverless/BargeCoachService';
import { SupervisorPrivateToggle } from '../SupervisorPrivateModeButtonComponent';

jest.mock('react-redux', () => ({
    useSelector: () => ({
      barge: true,
      coaching: true,
      privateMode: true
    }),
    useDispatch: () => jest.fn()
  }));
  jest.mock('../../../utils/sync/Sync', () => ({
    SyncDoc:{
        initSyncDoc:jest.fn()
    }
}))

  describe('Supervisor private button',()=>{
    it('renders the snapshot correctly',()=>{
        const mockTask = {
            conference: {
              conferenceSid: "CFxxxxxx",
              participants: [{
                participantType: 'worker',
                workerSid: 'flex-selector-test-id',
                callSid: 'CAxxxx'
              }],
              source: {
                channelParticipants: [
                  {
                    participantSid: 'PAxxxx',
                    type: 'supervisor',
                    mediaProperties: {
                      status: 'joined'
                    },
                    routingProperties: {
                      workerSid: 'flex-selector-test-id'
                    }
                  }
                ]
              }
            }
          }
          const wrapper = render(
            <SupervisorPrivateToggle task={mockTask} />
          )
          expect(wrapper).toMatchSnapshot();
    })
    it('call coachHandleClick successfully', async () => {
      const user = userEvent.setup();
        const mockTask = {
          conference: {
            conferenceSid: "CFxxxxxx",
            participants: [{
              participantType: 'worker',
              workerSid: 'flex-selector-test-id',
              callSid: 'CAxxxx'
            }],
            source: {
              channelParticipants: [
                {
                  participantSid: 'PAxxxx',
                  type: 'supervisor',
                  mediaProperties: {
                    status: 'joined'
                  },
                  routingProperties: {
                    workerSid: 'flex-selector-test-id'
                  }
                }
              ]
            }
          }
        }
        const { getByTestId } = render(
          <SupervisorPrivateToggle task={mockTask} />
        )
        const togglePrivateBtn = getByTestId('togglePrivateBtn');
        expect(togglePrivateBtn).toBeEnabled();
        await user.click(togglePrivateBtn);
        //expect(CallbackService.callCustomerBack).toHaveBeenCalled();
      });
  })