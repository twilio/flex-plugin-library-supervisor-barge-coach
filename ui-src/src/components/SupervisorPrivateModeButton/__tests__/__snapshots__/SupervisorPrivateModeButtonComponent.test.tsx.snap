// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Supervisor private button renders the snapshot correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <style>
      
    table.disable-selection {
      -webkit-touch-callout: none;
      -webkit-user-select: none; 
      -khtml-user-select: none; 
      -moz-user-select: none; 
      -ms-user-select: none; 
      user-select: none;
    }
  
    .disable-selection span::selection{
      background-color: transparent;
    }
    .disable-selection br::selection{
      background-color: transparent;
    }
  
    </style>
    <div>
      <div
        class="css-1dhdhv0"
        data-paste-element="FLEX"
      >
        <div
          class="css-tfro19"
          data-paste-element="SUPERVISOR_PRIVATE_BUTTON_BOX"
        >
          <div
            class="css-ci45q"
            data-paste-element="SUPERVISOR_PRIVATE_BUTTON_BOX_CHILD"
          >
            <button
              icon="EyeBold"
              title="Disable Private Mode"
              variant="secondary"
            />
          </div>
          <div
            class="css-roynbj"
            data-paste-element="SUPERVISOR_PRIVATE_BUTTON_BOX_CHILD"
          >
            Privacy On
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-1dhdhv0"
      data-paste-element="FLEX"
    >
      <div
        class="css-tfro19"
        data-paste-element="SUPERVISOR_PRIVATE_BUTTON_BOX"
      >
        <div
          class="css-ci45q"
          data-paste-element="SUPERVISOR_PRIVATE_BUTTON_BOX_CHILD"
        >
          <button
            icon="EyeBold"
            title="Disable Private Mode"
            variant="secondary"
          />
        </div>
        <div
          class="css-roynbj"
          data-paste-element="SUPERVISOR_PRIVATE_BUTTON_BOX_CHILD"
        >
          Privacy On
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
